### PR TITLE
Correctly apply PNG palette when building ImageBase through Pillow.

### LINKED ISSUE #14293: imshow() producing "inverted" colors since 3.0.3
### Bug report

**Bug summary**

Since 3.0.3 imshow() produces "inverted" colors, instead of just including the image as is.

**Code for reproduction**

```python
from PIL import Image
import matplotlib.pyplot as plt

img = Image.open("map.png")
plt.imshow(img)
plt.savefig("plot.png")
```

with this image as input:
map

**Actual outcome**
plot

**Expected outcome**
plot2
(until 3.0.2)

**Matplotlib version**

  * Operating system: win 7/10
  * Matplotlib version: 3.0.3, 3.1.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: 3.7.2
  * Jupyter version (if applicable):
  * Other libraries: Pillow (6.0.0)

matplotlib installed from pip

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Closes #14293.
A test would be nice but I would prefer *not* adding a test image (as usual), so I'd need to think more about the best way to do it.  In the meantime, the fact that this closes #14293 can be checked manually...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
Ah, that makes a lot more sense that my guess of pulling off just one of the channels.

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -667,6 +667,13 @@ def set_data(self, A):
         ----------
         A : array-like
```
Comment (dstansby) [inline lib/matplotlib/image.py]:
```suggestion
        A : array-like or `PIL.Image.Image`
```

Comment (dstansby) [inline lib/matplotlib/image.py]: Could add here that this now also accepts a `PIL.Image.Image` type?

Comment (anntzer) [reply lib/matplotlib/image.py]: Technically, PIL images *are* array-likes (in the sense that `asarray`, for example, works on them).

Comment (timhoffm) [reply lib/matplotlib/image.py]: OTOH not every place that takes array-likes can take a PIL image. Maybe we need a definition of *array-like* (technically https://stackoverflow.com/questions/40378427/numpy-formal-definition-of-array-like-objects).

However, from a practical point of view, most places we use *array-like* do not reasonably take PIL images (even if they sometimes technically could). For the few places of dedicated image-related stuff, I would be explicit and use `array-like or PIL.Image.Image`. Not every user will know that an PIL Image is technically an array-like).

Comment (anntzer) [reply lib/matplotlib/image.py]: fair enough, edited accordingly

**Comment by jklymak** [comment]
If this had a test we would have caught it breaking in the first place.

**Comment by anntzer** [comment]
Sure... added a test, and found #14340 at the same time...

File: lib/matplotlib/tests/test_image.py [hunk 2]
```diff
@@ -117,11 +117,16 @@ def test_image_python_io():
...
-    fig_ref.subplots().imshow(np.asarray(img))
+    style.use("default")
+    PIL = pytest.importorskip("PIL")
+    png_path = Path(__file__).parent / "baseline_images/pngsuite/basn3p04.png"
+    tiff_path = Path(__file__).parent / "baseline_images/test_image/uint16.tif"
+    axs = fig_test.subplots(2)
+    axs[0].imshow(PIL.Image.open(png_path))
+    axs[1].imshow(PIL.Image.open(tiff_path))
+    axs = fig_ref.subplots(2)
+    axs[0].imshow(plt.imread(str(png_path)))
```
Comment (jklymak) [inline lib/matplotlib/tests/test_image.py]: Looks good - why is this "str" here?

Comment (anntzer) [reply lib/matplotlib/tests/test_image.py]: because imread() doesn't support Path inputs *for png only*.

**Pull request merged**